src/views/tutorial-view/server.ts

```diff
@@ -0,0 +1,72 @@
+import { Product as ProductContext } from 'types/products'
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import { getTutorial } from 'lib/learn-client/api/tutorial'
+import {
+  ProductOption,
+  TutorialFullCollectionCtx as ClientTutorial,
+} from 'lib/learn-client/types'
+import { stripUndefinedProperties } from 'lib/strip-undefined-props'
+import { splitProductFromFilename } from './helpers'
+
+// @TODO just a stub - adjust page props interface
+export interface TutorialPageProps {
+  tutorial: ClientTutorial
+  product: Pick<ProductContext, 'slug'>
+}
+
+export async function getTutorialPageProps(
+  product: ProductOption,
```

Comment:
Two questions here:

- Could the `product` variables be renamed to `productSlug` where a slug is expected and not a `Product` object? We have been distinguishing the two with clearer variable names in addition to TypeScript types (`Product` and `ProductSlug`).
- Have you considered using `ProductSlug` (from `src/types/products.ts`) here instead of `ProductOption`? This is more general TypeScript question, probably comes down to `type` vs. `enum`.

Reply:
Sure! I can update this name. I'd prefer to continue with the enum pattern for now to maintain parity between what is currently in Learn. If there's room to abstract and share later, we can join the two.

Reply:
> I'd prefer to continue with the enum pattern for now to maintain parity between what is currently in Learn

Sounds good! Totally reasonable.

> we lean on enums heavily in our typing right now

We've wanted to make improvements to the `Product` type since it encompasses so much, but have been waiting until we understand its usage more widely before abstracting. I wonder how `enums` could play into that. I haven't used them a ton yet and don't know much about their tradeoffs with other approaches. Guess this is more of a note to self to get to learning about them! I'm looking forward to seeing how y'all have used them. 😁